core/tests/puppeteer-acceptance-tests/user-utilities/exploration-creator-utils.ts

```diff
@@ -13,146 +13,623 @@
-    await this.type(richTextAreaField, `${explorationTitle}`);
-    await this.clickOn(saveContentButton);
+  async updateCardName(cardName: string): Promise<void> {
+    await this.clickOn(editCardNameButton);
+    await this.waitForElementToBeClickable(editCardNameBox);
+    await this.page.click(editCardNameBox, {clickCount: 3});
+    await this.page.keyboard.press('Backspace');
+    await this.type(editCardNameBox, cardName);
+    await this.clickOn(introSubmitButton);
+    showMessage('Card name updated successfully.');
+  }
 
+  /**
+   * This function clears previous intro text and
+   * add a new intro text.
+   */
+  async updateExplorationIntroText(introtext: string): Promise<void> {
+    await this.clickOn(explorationIntroBoxOpenerSection);
+    await this.clickOn(explorationIntroBox);
+    await this.waitForElementToBeClickable(explorationIntroBox);
+    await this.page.click(explorationIntroBox, {clickCount: 3});
+    await this.page.keyboard.press('Backspace');
+    await this.type(explorationIntroBox, introtext);
+    await this.clickOn(introTitleSubmitButton);
+    showMessage('Intro text updated successfully.');
+  }
+
+  async addEndInteraction(): Promise<void> {
+    await this.page.waitForNavigation({waitUntil: 'networkidle0'});
     await this.clickOn(addInteractionButton);
-    await this.clickOn(interactionEndExplorationInputButton);
+    await this.clickOn(endExplorationInteractionOption);
     await this.clickOn(saveInteractionButton);
-    await this.page.waitForSelector('.customize-interaction-body-container', {
-      hidden: true,
+    showMessage('End interaction has been added successfully.');
+  }
+  /**
+   * This function open settings tab.
+   * Note->It also opens all the dropdowns present
+   * in the setting tab for mobile view port.
+   */
+  async goToSettingsTab(): Promise<void> {
+    if (this.isViewportAtMobileWidth()) {
+      await this.page.waitForSelector(navBarOpenerIcon, {visible: true});
+      await this.clickOn(navBarOpenerIcon);
+      await this.clickOn(optionsDropDown);
+      await this.clickOn(mobileSettingsBar);
+      /**
+       * Open all dropdowns because by default
+       * all dropdowns are closed.
+       */
+      await this.clickOn(basicSettingsDropDown);
+      await this.clickOn(advanceSettingsDropDown);
+      await this.clickOn(rolesSettingsDropDown);
+      await this.clickOn(voiceArtistSettingsDropDown);
+      await this.clickOn(permissionSettingsDropDown);
+      await this.clickOn(feedbackSettingsDropdown);
+      await this.clickOn(explorationControlsSettingsDropDown);
+    } else {
+      await this.clickOn(settingsTab);
+    }
+    showMessage('Settings tab is opened successfully.');
+  }
+
+  /**
+   * This function deletes the previous written
+   * title and updates the new title.
+   */
+  async updateTitleTo(title: string): Promise<void> {
+    await this.waitForElementToBeClickable(addTitleBar);
+    await this.page.click(addTitleBar, {clickCount: 3});
```

Comment:
Already mentioned in earlier comment.
As you said to make each function self independent and robust.

Reply:
See previous comment on how to simplify this.

Additionally to the previous suggestion, you could probably also combine the triple-click and backspace into a single function `this.highlightAndDeleteText()`, since it gets reused.

Reply:
@seanlip I think` clearAllText()` will be more clear.

Reply:
Done